assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -9,38 +9,48 @@ interface State {
 
 interface Context {
 	woocommerce: {
-		productGallery: { numberOfThumbnails: number };
+		selectedImage: string;
+		imageId: string;
 	};
 }
 
 interface Selectors {
 	woocommerce: {
-		productGallery: {
-			numberOfPages: ( store: unknown ) => number;
-		};
+		isSelected: ( store: unknown ) => boolean;
+	};
+}
+
+interface Actions {
+	woocommerce: {
+		handleClick: ( context: Context ) => void;
```

Comment:
I'm a bit uncomfortable with the lack of specificity here. `selectedImage`, `imageId`, `isSelected`, `handleClick` seem all very generic, and they are located inside of `woocommerce`.

This seems to be very vulnerable when it comes to potential conflicts (especially, when more blocks get converted/built using Interactivity API).

Or am I missing something?

Reply:
>This seems to be very vulnerable when it comes to potential conflicts (especially, when more blocks get converted/built using Interactivity API).

It isn't an issue here.  All these functions are available only in that Product Gallery context since we set them via the `wc-data-context`.

Reply:
Ok, this is valuable info 🙏